plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/util/ElkUtil.java

```diff
@@ -396,6 +398,112 @@ public static void applyConfiguredNodeScaling(final ElkNode node) {
             }
         }
     }
+    
+    /**
+     * Computes the area occupied by this node's layout and stores the values in {@link CoreOptions#CHILD_AREA_WIDTH} 
+     * and {@link CoreOptions#CHILD_AREA_HEIGHT}.
+     * @param node the node whose child area should be computed
+     */
+    public static void computeChildAreaDimensions(final ElkNode node) {
+        // TODO: for some stuff like labels and maybe edges it might only be necessary
+        //       to check the max sides, because positioning of the layout already deals with
+        //       the min case, but better safe than sorry
+        
+        double minX = Double.MAX_VALUE;
```

Comment:
For a double Double.INFINITY is often used instead to signal if nothing was changed.